app/parse.py

```
@@ -8,5 +13,47 @@ class Course:
     duration: str
 
 
+COURSE_FIELDS = [field.name for field in fields(Course)]
+
+
+def parse_single_course(card: BeautifulSoup) -> Course:
+    content = card.find("div", class_="ProfessionCard_content__mPiVi")
+    return Course(
+        name=content.find(
+            "h3", class_="ProfessionCard_title__m7uno"
+        ).text,
+        short_description=content.find(
+            "p", class_="ProfessionCard_description__K8weo"
+        ).text,
+        duration=content.find(
+            "p", class_="ProfessionCard_duration__13PwX")
+        .text
+    )
+
+
 def get_all_courses() -> list[Course]:
-    pass
+    response = requests.get(BASE_URL)
```

The network call uses requests.get without timeout or error handling. If the request times out or returns a non-200 status, this will raise later when you try to use response.text (or silently proceed with a bad response). Wrap requests.get with try/except for requests.RequestException, pass a timeout (e.g., timeout=10), and call response.raise_for_status() to ensure non-200 responses raise. On failure, return an empty list or raise a clear exception so callers can handle it.